src/operator/tensor/broadcast_reduce_op.h

```diff
@@ -339,6 +339,49 @@ inline void BroadcastReduceShapeCompact(const TShape& big, const TShape& small,
   }
 }
 
+inline bool SumOpForwardInferStorageType(const nnvm::NodeAttrs& attrs,
+                                         const int dev_mask,
+                                         DispatchMode* dispatch_mode,
+                                         std::vector<int>* in_attrs,
+                                         std::vector<int>* out_attrs) {
+  CHECK_EQ(in_attrs->size(), 1U);
+  CHECK_EQ(out_attrs->size(), 1U);
+  const ReduceAxesParam& param = nnvm::get<ReduceAxesParam>(attrs.parsed);
+  int& in_stype = in_attrs->at(0);
```

Comment:
Use `int` or `const int`. No need to use reference for primitive types in C++.

Reply:
I can use the int for in_stype. I cannot use it for out_stype though because I need a reference to modify out_attrs. This breaks otherwise.